CalibCalorimetry/HcalTPGAlgos/src/LutXml.cc

```diff
@@ -296,37 +298,27 @@ int LutXml::test_access(std::string filename) {
   return 0;
 }
 
-HcalSubdetector LutXml::subdet_from_crate(int crate_, int eta, int depth) {
-  HcalSubdetector result;
-  // HBHE: 0,1,4,5,10,11,14,15,17
-  // HF: 2,9,12
-  // HO: 3,6,7,13
+HcalSubdetector LutXml::subdet_from_crate(int crate_, int slot, int fiber) {
+  // HBHE: 0,1,4,5,10,11,14,15,17 (+20)
+  // HF: 2,9,12  (+20)
+  // HO: 3,6,7,13,18  (+20)
   int crate = crate_ < 20 ? crate_ : crate_ - 20;
-
   if (crate == 2 || crate == 9 || crate == 12)
-    result = HcalForward;
-  else if (crate == 3 || crate == 6 || crate == 7 || crate == 13)
-    result = HcalOuter;
+    return HcalForward;
+  else if (crate == 3 || crate == 6 || crate == 7 || crate == 13 || crate == 18)
+    return HcalOuter;
   else if (crate == 0 || crate == 1 || crate == 4 || crate == 5 || crate == 10 || crate == 11 || crate == 14 ||
            crate == 15 || crate == 17) {
-    if (eta < 16)
-      result = HcalBarrel;
-    else if (eta > 16)
-      result = HcalEndcap;
-    else if (eta == 16 && depth < 3)
-      result = HcalBarrel;
-    else if (eta == 16 && depth >= 3)
-      result = HcalEndcap;
-    else {
-      edm::LogError("LutXml") << "Impossible to determine HCAL subdetector!!!";
-      exit(-1);
-    }
-  } else {
-    edm::LogError("LutXml") << "Impossible to determine HCAL subdetector!!!";
-    exit(-1);
+    if (slot % 3 == 1)
+      return HcalBarrel;
+    else if (slot % 3 == 0)
+      return HcalEndcap;
+    else if (fiber < 12)
+      return HcalBarrel;
+    else
+      return HcalEndcap;
   }
-
-  return result;
+  return HcalOther;
```

Comment:
Why is HcalOther being used here? Doesn't HcalOther correspond to calibration channels, for which there is no sensible definition of a trigger tower sum?

Reply:
in this case it's just anything Other than normal channel. It should not normally get to this. Changed to HcalEmpty and added a warning.